node-hub/opencv-plot/main.py

```diff
@@ -113,50 +134,29 @@ def main():
                     "width": np.uint32(arrow_image["width"].as_py()),
                     "height": np.uint32(arrow_image["height"].as_py()),
                     "channels": np.uint8(arrow_image["channels"].as_py()),
-                    "data": arrow_image["data"].values.to_numpy().astype(np.uint8)
+                    "data": arrow_image["data"].values.to_numpy().astype(np.uint8),
                 }
 
-                plot.frame = np.reshape(image["data"], (image["height"], image["width"], image["channels"]))
+                plot.frame = np.reshape(
+                    image["data"], (image["height"], image["width"], image["channels"])
+                )
 
                 plot_frame(plot)
                 if not RUNNER_CI:
                     if cv2.waitKey(1) & 0xFF == ord("q"):
                         break
-                else:
-                    break  # break the loop for CI
-
             elif event_id == "bbox":
                 arrow_bbox = event["value"][0]
                 plot.bboxes = {
                     "bbox": arrow_bbox["bbox"].values.to_numpy().reshape(-1, 4),
                     "conf": arrow_bbox["conf"].values.to_numpy(),
                     "names": arrow_bbox["names"].values.to_numpy(zero_copy_only=False),
                 }
-
-                plot_frame(plot)
-                if not RUNNER_CI:
-                    if cv2.waitKey(1) & 0xFF == ord("q"):
-                        break
```

Comment:
When sending a image flux that's okay. But, you always receive bboxes after the frame it was calculated on. So not plotting bboxes when receiving them could create a lag

Reply:
So I understand that there might be a perceived lag, but I think it might be easier to have a scalable to many input logic despite loosing in performance rather than choosing a non-scalable logic for performance.

This way we can also add other marker input without too much cost.